src/ngResource/resource.js

```diff
@@ -625,6 +703,7 @@ angular.module('ngResource', ['ng']).
                 }
               }
 
+              value.$cancelRequest = cancelRequest;
```

Comment:
we can just set this to `noop` here and then we don't need to hold on to the promise up above.